components/kyma-environment-broker/internal/dto.go

```diff
@@ -72,7 +73,35 @@ const (
 	Openstack       CloudProvider = "OpenStack"
 )
 
+type AutoScalerParameters struct {
+	AutoScalerMin  *int `json:"autoScalerMin"`
+	AutoScalerMax  *int `json:"autoScalerMax"`
+	MaxSurge       *int `json:"maxSurge"`
+	MaxUnavailable *int `json:"maxUnavailable"`
+}
+
+// FIXME: this is a makeshift check until the provisioner is capable of returning error messages
+// https://github.com/kyma-project/control-plane/issues/946
+func (p AutoScalerParameters) Validate() error {
+	if p.AutoScalerMin == nil && p.AutoScalerMax == nil {
+		return nil
+	}
+	if p.AutoScalerMin == nil || p.AutoScalerMax == nil {
+		missing := "Min"
+		if p.AutoScalerMax == nil {
+			missing = "Max"
+		}
+		return fmt.Errorf("When setting Auto Scaler Min/Max you need to provide both, missing: %v", missing)
```

Comment:
because it is technically possible at the moment to provide only min or max which makes it much harder to validate because even the operation in the db won't specify any value. At that point, KEB was relying on provisioner/gardener setting defaults.

Reply:
we should allow to update only min, or only max. If wrong values we should say why. Keb should take values from db and send to provisioner, we should log that user has changed only e.g. min.

Reply:
thanks for the input, I will try to figure something out then, but it is unfortunate that some defaulting is done in KEB, some in provisioner and some in gardener, it makes the design a little more convoluted.

The problem here is there is nothing in the database for the following scenario. If you set only `"autoScalerMin": 11` during provisioning, the operation in db would contain these values
```
broker=> select id, type, state, finished_stages, provisioning_parameters from operations where instance_id='test-jw40';
... "autoScalerMin":11,"autoScalerMax":null,"maxSurge":null,"maxUnavailable":null ...
```
and be forever in pending state because provisioner would default `autoScalerMax` to 10.
My naive idea is to take the provisioner hardcoded max as default in KEB too and if the user doesn't set max, set it to 10. If user sets min higher than 10, make max == min.

Reply:
I was wrong, it was explained to me that it is a plan defaulting that is modifying the value in the operations queue. Implemented reuse of this logic in https://github.com/kyma-project/control-plane/pull/958/commits/094b8ccdf71f6a07da372246ac5e72a832dbb55a